app/src/main/java/com/guru/bluetooth/FileSenderActivity.kt

```diff
@@ -17,27 +17,27 @@ class FileSenderActivity : AppCompatActivity() {
 
         val device: BluetoothDevice? = getBluetoothDeviceFromIntent(intent)
         val fileURI: String? = getFileUriFromIntent(intent)
-        val sendingResult = device?.let { startBluetoothClient(it, fileURI!!) }
-        updateSendingResult(sendingResult == true)
+        val sendingResult: Boolean? = device?.let { startBluetoothClient(it, fileURI) }
+        sendingResult?.let { updateSendingResult(it) }
     }
 
     private fun getBluetoothDeviceFromIntent(intent: Intent): BluetoothDevice? {
         return intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE)
     }
 
     private fun getFileUriFromIntent(intent: Intent): String? {
-        return intent.getStringExtra(AlarmClock.EXTRA_MESSAGE)
+        return intent.getStringExtra(EXTRA_MESSAGE)
     }
 
-    private fun startBluetoothClient(device: BluetoothDevice, fileUri: String): Boolean {
-        return BluetoothConnectionService().startClient(device, fileUri)
+    private fun startBluetoothClient(device: BluetoothDevice, fileUri: String?): Boolean? {
```

Comment:
the return type of function startBluetoothClient is Boolean?
is that expected, we should return a non-null value

Reply:
Yes, the return type of the startBluetoothClient function is Boolean?, which is a nullable Boolean type. This means that the function can return a Boolean value or a null value.